.github/workflows/tests.yml

```diff
@@ -0,0 +1,202 @@
+        run: |
+          sudo apt install tox
+
+      - name: Run linter
+        run: |
+          tox -e pep8
+
+  unit:
+    name: Unit Tests
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.10", "3.12"]
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install dependencies
+        run: |
+          sudo apt install tox
+
+      - name: Run Unit tests
+        run: |
+          tox -e py3
+
+  coverage:
+    name: Coverage
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Install dependencies
+        run: |
+          sudo apt install tox
+
+      - name: Run coverage
+        run: |
+          tox -e cover
+
+  build:
+    name: Build charm
+    needs:
+      - lint
+      - unit
+      - coverage
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Install dependencies
+        run: |
+          sudo apt install tox
+          sudo snap install charmcraft --channel 3.x/stable --classic
+          sudo snap install lxd
+          sudo snap set lxd daemon.group=adm
+          sudo lxd init --auto
+
+      - name: Clear FORWARD firewall rules
+        run: |
+          # Docker can inject rules causing firewall conflicts
+          sudo iptables -P FORWARD ACCEPT  || true
+          sudo ip6tables -P FORWARD ACCEPT || true
+          sudo iptables -F FORWARD  || true
+          sudo ip6tables -F FORWARD || true
+
+      - name: Build charm
+        run: |
+          tox -e build
+
+      - name: Store charm as artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: ovn-chassis-charm
+          path: ./ovn-chassis_amd64.charm
+          retention-days: 7
+
+  parse_tags:
+    name: Parse tags in PR commits
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: read
+    outputs:
+      zaza_pr: ${{ steps.find-zaza-pr.outputs.zaza_pr }}
+      
+    steps:
+      - name: Get PR commits
+        id: 'get-pr-commits'
+        uses: tim-actions/get-pr-commits@master
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Find requested functional test PR
+        id: 'find-zaza-pr'
+        run: |
+          # Note(mkalcok): The command below cleanely transfers data from Github
+          #                variable to bash variable without interpreting curly
+          #                braces or losing quotations.
+          COMMIT_DATA=$(
+          cat <<'EOF'
+          ${{ steps.get-pr-commits.outputs.commits }}
+          EOF
+          )
+
+          ZAZA_TAG="Func-test-pr: "
+          FUNC_PR=$(jq -r '.[].commit.message' <<< "$COMMIT_DATA" | grep "^${ZAZA_TAG}" | tail -1)
```

Comment:
Should we allow multiple variations of capitalization of `func-test-pr` tag? E.g. would adding a `-i` to grep make our users lives easier?

Reply:
right, the `ZAZA_TAG` variable is also used few lines below to remove prefix from string, but I'm sure there's a way to do that case insensitively too.
👍